protocol/messenger_backup.go

```diff
@@ -224,12 +236,49 @@ func (m *Messenger) BackupData(ctx context.Context) (uint64, error) {
 		pb := backupDetailsOnly()
 		pb.WatchOnlyAccountDetails.DataNumber = uint32(i + 1)
 		pb.WatchOnlyAccount = d.WatchOnlyAccount
+		// TODO find a way to get all settings
+		// fullBackup.WatchOnlyAccount = append(fullBackup.WatchOnlyAccount, d.Keypair)
 		err = m.encodeAndDispatchBackupMessage(ctx, pb, chat.ID)
 		if err != nil {
 			return 0, err
 		}
 	}
 
+	if m.config.featureFlags.EnableLocalBackup {
+		// TODO put file in a constant
+		path := filepath.Join(m.config.backupConfig.DataDir, "user_data.bkp")
+
+		if err := os.MkdirAll(m.config.backupConfig.DataDir, 0700); err != nil {
+			return 0, err
+		}
+
+		file, err := os.Create(path)
+		if err != nil {
+			return 0, err
+		}
+		defer file.Close()
+
+		mashalledMessage, err := proto.Marshal(fullBackup)
+		if err != nil {
+			return 0, err
+		}
+
+		messageSpec, err := m.encryptor.BuildDHMessage(m.identity, &m.identity.PublicKey, mashalledMessage)
```

Comment:
This will soon be encapsulated within the messaging facade. There's no need to involve the messaging machinery, such as DH and Protocol Message, for local backups.

Encrypting `protobuf.Backup{}` with AES would be way simpler.

Reply:
No problem. I won't lie, I mostly improvised here because I don,t have experience with the encryption we have available.

Could you point me the function I should use instead please :pray:

Reply:
@osmaczko I think I figured it out. Please review again and let me know if that's what you had in mind. The test still passes with this new encryption method. It's definitely cleaner this way.